domain/consensus/timelock_CSV_test.go

```diff
@@ -0,0 +1,524 @@
+		if err != nil {
+			t.Fatalf("Error creating transactionThatSpentTheLockedOutput: %v", err)
+		}
+		// Add a block that contains a transaction that spends the locked output before the time, and therefore should be failed.
+		_, _, err = testConsensus.AddBlock([]*externalapi.DomainHash{blockEHash}, nil,
+			[]*externalapi.DomainTransaction{transactionThatSpentTheLockedOutput})
+		if err == nil || !errors.Is(err, ruleerrors.ErrUnfinalizedTx) {
+			t.Fatalf("Expected block to be invalid with err: %v, instead found: %v", ruleerrors.ErrUnfinalizedTx, err)
+		}
+		emptyCoinbase := externalapi.DomainCoinbaseData{
+			ScriptPublicKey: &externalapi.ScriptPublicKey{
+				Script:  nil,
+				Version: 0,
+			},
+		}
+		var tipHash *externalapi.DomainHash
+		blockE, err := testConsensus.GetBlock(blockEHash)
+		if err != nil {
+			t.Fatalf("Failed to get blockE: %v", err)
+		}
+		timeStampBlockE := blockE.Header.TimeInMilliseconds()
+		stagingArea := model.NewStagingArea()
+		// Make sure the time limitation has passed.
+
+		lockTimeTarget := uint64(blockE.Header.TimeInMilliseconds()) + timeToWait*constants.SequenceLockTimeGranularity
+		for i := int64(0); ; i++ {
+			tipBlock, err := testConsensus.BuildBlock(&emptyCoinbase, nil)
+			if err != nil {
+				t.Fatalf("Error creating tip using BuildBlock: %v", err)
+			}
+			blockHeader := tipBlock.Header.ToMutable()
+			blockHeader.SetTimeInMilliseconds(timeStampBlockE + i*constants.SequenceLockTimeGranularity)
+			tipBlock.Header = blockHeader.ToImmutable()
+			_, err = testConsensus.ValidateAndInsertBlock(tipBlock)
+			if err != nil {
+				t.Fatalf("Error validating and inserting tip block: %v", err)
+			}
+			tipHash = consensushashing.BlockHash(tipBlock)
+			pastMedianTime, err := testConsensus.PastMedianTimeManager().PastMedianTime(stagingArea, tipHash)
+			if err != nil {
+				t.Fatalf("Failed getting pastMedianTime: %v", err)
+			}
+			if uint64(pastMedianTime) > lockTimeTarget {
+				break
+			}
+		}
+		// Tries to spend the output that should be no longer locked
+		validBlock, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{tipHash}, nil,
+			[]*externalapi.DomainTransaction{transactionThatSpentTheLockedOutput})
+		if err != nil {
+			t.Fatalf("The block should be valid since the output is not locked anymore. but got an error: %v", err)
+		}
+		validBlockStatus, err := testConsensus.BlockStatusStore().Get(testConsensus.DatabaseContext(), stagingArea,
+			validBlock)
+		if err != nil {
+			t.Fatalf("Failed getting the status for validBlock: %v", err)
+		}
+		if !validBlockStatus.Equal(externalapi.StatusUTXOValid) {
+			t.Fatalf("The status of validBlock should be: %v, but got: %v", externalapi.StatusUTXOValid,
+				validBlockStatus)
+		}
+	})
+}
+
+// TestRelativeTimeOnCheckSequenceVerify verifies that if the relative target is set to be X DAA score,
+// and the absolute DAA score is X before having X DAA score more than the time the UTXO was mined, then the output will remain locked.
```

Comment:
> and the absolute DAA score is

There's no such things as "the absolute DAA score", there's "the actual DAA score" or something similar.

Reply:
Also the rest of the sentance is really hard to understand, please revise (Talk to me if need someone to brainstorm with)